dist2src/cli.py

```diff
@@ -132,19 +133,106 @@ def get_archive(gitdir):
     return stdout
 
 
-def _copy_files(
-    origin: str, dest: str, origin_dir: str, dest_dir: str, glob: str
-) -> None:
+@cli.command()
+@click.argument("path", type=click.Path(exists=True, file_okay=False))
+@log_call
+def run_prep(path):
+    """Run `rpmbuild -bp` in GITDIR.
+
+    PATH needs to be a dist-git repository.
     """
-    Copy all glob files from origin/origin_dir to dest/dest_dir
+    rpmbuild = sh.Command("rpmbuild")
+
+    with sh.pushd(path):
+        cwd = Path.cwd()
+        logger.debug(f"Running rpmbuild in {cwd}")
+        specfile_path = f"SPECS/{cwd.name}.spec"
+        setup = re.compile("^%setup ", re.MULTILINE)
+        with open(specfile_path, "r") as fp:
+            spec = fp.read()
+        spec, number_of_subs = setup.subn("%gitsetup ", spec)
+        if number_of_subs > 1:
+            raise RuntimeError("Wow! Multiple %setup macros in the spec file!")
+        if number_of_subs:
+            with open(specfile_path, "w") as fp:
+                fp.write(spec)
+        try:
+            stdout = rpmbuild(
+                "--nodeps",
+                "--define",
+                f"_topdir {cwd}",
+                "--define",
+                "__scm git",
+                "-bp",
+                specfile_path,
+            )
+        except sh.ErrorReturnCode as e:
+            for line in e.stderr.splitlines():
+                logger.debug(str(line))
+            raise
+
+    return stdout
+
+
+def get_build_dir(path: Path):
+    build_dirs = [d for d in (path / "BUILD").iterdir() if d.is_dir()]
+    if len(build_dirs) > 1:
+        raise RuntimeError(f"More than one directory found in {path}")
+    if len(build_dirs) < 1:
+        raise RuntimeError(f"No subdirectory found in {path}")
+    return build_dirs[0]
+
+
+@cli.command()
+@click.argument("path", type=click.Path(exists=True, file_okay=False))
+@log_call
+@click.pass_context
+def commit_all(ctx, path):
+    """Git add and commit all the changes in PATH.
+
+    Do this, b/c some %prep sections will modify source code with scripts,
+    and the current understanding is that we should capture these changes.
+    """
+    build_dir = get_build_dir(path)
+    repo = git.Repo(build_dir)
+    if repo.is_dirty():
+        ctx.invoke(stage, gitdir=build_dir)
+        ctx.invoke(commit, m="Various changes", gitdir=build_dir)
+
+
+@cli.command()
+@click.argument("source_dir", type=click.Path(exists=True, file_okay=False))
+@click.argument("dest_dir", type=click.Path(exists=True, file_okay=False))
+@click.argument("dest_branch", type=click.STRING)
+@log_call
+def pull_branch(source_dir, dest_dir, dest_branch):
+    """Run 'git -C DEST_DIR pull --ff-only SOURCE_DIR +master:DEST_BRANCH'.
```

Comment:
I'm having difficulties to understand **why** we do this. Can you shed more light on it either here or in README? Thanks

Reply:
Oh, actually this doc-string is outdated. Let me update it first.